features/admin.actions.v1/pages/actions.tsx

```diff
@@ -235,45 +270,64 @@ export const ActionTypesListingPage: FunctionComponent<ActionTypesListingPageInt
         );
 
     const actionTypesCardsInfo = (): ActionTypeCardInterface[] => {
-        return [
-            {
-                description: t("actions:types.preIssueAccessToken.description.shortened"),
-                disabled: isActionTypeDisabled(FeatureFlagConstants.FEATURE_FLAG_KEY_MAP
-                    .ACTIONS_TYPES_PRE_ISSUE_ACCESS_TOKEN),
-                featureStatusKey: FeatureFlagConstants.FEATURE_FLAG_KEY_MAP
-                    .ACTIONS_TYPES_PRE_ISSUE_ACCESS_TOKEN,
-                heading: t("actions:types.preIssueAccessToken.heading"),
-                icon: <KeyFlowIcon size="small" className="icon"/>,
-                identifier: ActionsConstants.PRE_ISSUE_ACCESS_TOKEN_URL_PATH,
-                route: isActionTypeDisabled(FeatureFlagConstants.FEATURE_FLAG_KEY_MAP
-                    .ACTIONS_TYPES_PRE_ISSUE_ACCESS_TOKEN) ? undefined :
-                    AppConstants.getPaths().get("PRE_ISSUE_ACCESS_TOKEN_EDIT")
-            },
-            {
-                description: t("actions:types.preUpdatePassword.description.shortened"),
-                disabled: isActionTypeDisabled(FeatureFlagConstants.FEATURE_FLAG_KEY_MAP
-                    .ACTIONS_TYPES_PRE_UPDATE_PASSWORD),
-                featureStatusKey: FeatureFlagConstants.FEATURE_FLAG_KEY_MAP.ACTIONS_TYPES_PRE_UPDATE_PASSWORD,
-                heading: t("actions:types.preUpdatePassword.heading"),
-                icon: <PadlockAsteriskFlowIcon size="small" className="icon"/>,
-                identifier: ActionsConstants.PRE_UPDATE_PASSWORD_URL_PATH,
-                route: isActionTypeDisabled(FeatureFlagConstants.FEATURE_FLAG_KEY_MAP
-                    .ACTIONS_TYPES_PRE_UPDATE_PASSWORD) ? undefined
-                    : AppConstants.getPaths().get("PRE_UPDATE_PASSWORD_EDIT")
-            },
-            {
-                description: t("actions:types.preUpdateProfile.description.shortened"),
-                disabled: isActionTypeDisabled(FeatureFlagConstants
-                    .FEATURE_FLAG_KEY_MAP.ACTIONS_TYPES_PRE_UPDATE_PROFILE),
-                featureStatusKey: FeatureFlagConstants.FEATURE_FLAG_KEY_MAP.ACTIONS_TYPES_PRE_UPDATE_PROFILE,
-                heading: t("actions:types.preUpdateProfile.heading"),
-                icon: <ProfileFlowIcon size="small" className="icon"/>,
-                identifier: ActionsConstants.PRE_UPDATE_PROFILE_URL_PATH,
-                route: isActionTypeDisabled(FeatureFlagConstants.FEATURE_FLAG_KEY_MAP
-                    .ACTIONS_TYPES_PRE_UPDATE_PROFILE) ? undefined
-                    : AppConstants.getPaths().get("PRE_UPDATE_PROFILE_EDIT")
-            },
-            {
+
+        if (!enabledActionTypes) return;
+        const actionTypeCards: ActionTypeCardInterface[] = [];
+
+        for (const actionType of enabledActionTypes) {
```

Comment:
instead of pushing card info conditionally, shall we maintain all card info in the array and then filter out the disabled actions?